src/libostree/ostree-repo.h

```diff
@@ -687,6 +687,28 @@ gboolean      ostree_repo_write_archive_to_mtree (OstreeRepo                   *
                                                   GError                      **error);
 
 /**
+ * OstreeRepoImportArchiveTranslatePathname:
+ * @repo: Repo
+ * @file_type: File type
+ * @src_path: Path in the archive
+ * @user_data: User data
+ *
+ * Possibly change a pathname while importing an archive. If %NULL is returned,
+ * then @src_path will be used unchanged.  Otherwise, return a new pathname which
+ * will be freed via `g_free()`.
+ *
+ * This pathname translation will be performed *before* any processing from an
+ * active `OstreeRepoCommitModifier`. Will be invoked for all directory and file
+ * types, first with outer directories, then their sub-files and directories.
+ *
+ * Since: 2017.11
+ */
+typedef char *(*OstreeRepoImportArchiveTranslatePathname) (OstreeRepo     *repo,
+                                                           GFileType       file_type,
```

Comment:
I did that originally, but then we'd need to `#include <sys/stat.h>` in `ostree.h`.  Which...is probably fine but takes us a bit farther down the unix-only path.  OK, so libglnx already makes us linux-specific...yeah, I'll make that change back.